src/Oscoin/Storage/Block/Abstract.hs

```diff
@@ -8,36 +8,60 @@
 
 module Oscoin.Storage.Block.Abstract
     ( BlockStore(..)
+    , FullBlockStoreAPI
+    , GetBlocksQuery(..)
 
     , hoistBlockStore
-    , noValidation
     , insertBlocksNaive
     , isNovelBlock
+
+    -- * Internals
+    , BlockStoreInternals(..)
+    , hoistBlockStoreInternals
     ) where
 
 import           Oscoin.Prelude
 
-import           Oscoin.Consensus.Types (Validate)
 import           Oscoin.Crypto.Blockchain (TxLookup)
 import           Oscoin.Crypto.Blockchain.Block
 import           Oscoin.Crypto.Hash (Hashed)
+import           Oscoin.Time.Chrono (NewestFirst, OldestFirst)
 
--- | An handle over a block storage backend.
-data BlockStore c tx s m = BlockStore
+-- The full 'BlockStore' API is the composition of its public and private parts.
+type FullBlockStoreAPI c tx s m =
+    ( BlockStore c tx s m
+    , BlockStoreInternals c tx s m
+    )
```

Comment:
I was thinking about the naming here. How about the following?
~~~haskell
type BlockStore = (BlockStoreModifier, BlockStoreReader)
~~~

Reply:
That's a reasonable suggestion (which will again have me to amend virtually every call site in the codebase 😁 ). I do agree names are important though; let's see what the other folks say before I embark into this journey. The rationale behind calling things the current way was to minimise disruption; furthermore, the `Internals` thing was not meant to be used anywhere (if not in test code) and its naming was predated from [other libraries](http://hackage.haskell.org/package/network-transport-inmemory-0.5.2/docs/Network-Transport-InMemory-Internal.html#v:createTransportExposeInternals) , but I am happy to stick with what we think is best 😉

Reply:
(addendum: personal preference -- if we go with your suggestion, I think I like `BlockStoreWriter` better 😁 )